src/domainobjects/price.py

```diff
@@ -3,29 +3,47 @@
 import random
 
 class Price(Generatable):
+    """ Class to generate prices. Generate method will generate a set amount
+    of prices. """
+
+    def generate(self, record_count, start_id):
+        """ Generate a set number of prices
+
+        Parameters
+        ----------
+        record_count : int
+            Number of prices to generate
+        start_id : int
+            Starting id to generate from
+
+        Returns
+        -------
+        List
+            Containing 'record_count' prices
+        """
 
-    def generate(self, record_count, custom_args):
-        config = self.get_object_config()
-        records_per_file = config['max_objects_per_file']
-        file_num = 1
         records = []
+        self.instruments = self.retrieve_records('instruments')
+
+        for _ in range(start_id, start_id+record_count):
+            record = self.generate_record()
+            records.append(record)
+
+        return records
 
-        database = self.get_database()
-        instruments = database.retrieve('instruments')
-                
-        for i in range(1, record_count+1):
-            instrument = random.choice(instruments)      
-            records.append({
+    def generate_record(self):
+        """ Generate a single price
+
+        Returns
+        -------
+        dict
+            A single price object
+        """
+
+        instrument = self.get_random_instrument()
+        return {
                 'ric': instrument['ric'],
                 'price': self.generate_random_decimal(),
-                'curr': self.generate_currency(),
+                'currency': self.generate_currency(),
```

Comment:
currency should come from the security itself as well. I don't have a problem with currency and country of issue being duplicated in exchange, security and price since we need to link the data together to make it realistic.